### PR TITLE
Enchanted - Weapons for Magiclysm

### PR DESCRIPTION
**Pull request by foulman**
```SUMMARY: Mods "Basic magical weapons and enchanted item crafting framework for Magiclysm"```

Included is the most basic magical weapons, weapons +1 and +2. 
This covers almost all melee weapons, though I'm sure some were missed. 
The improvements from a standard weapon to a +1/+2 one are an equivilent increase in to_hit, a 10% increase in damage per +, and a 10% decrease in weight, thus improving attack speeds.

The framework for the rest of crafting adds an additional tab on the crafting screen at the end, called Enchanted.
For the time being this is primarily a placeholder, but there will be recipes flowing into it as recipes are created to go with content items.
Players will be able to craft their own +1/+2, and beyond, weapons and assorted magical gear.

Enchanting items will be very time consuming, as one would imagine. 
Be prepared to spend 24, 48, or even 72 hours and more of time working on items.

### CODE REVIEW

File: data/mods/Magiclysm/enchanted/enchanted_melee.json [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+    "proportional": { "price": 3.0, "bashing": 1.1, "cutting": 1.1 },
+    "relative": { "to_hit": 1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "cudgel_plus_two",
+    "copy-from": "cudgel",
+    "looks_like": "cudgel",
+    "name": "cudgel +2",
+    "proportional": { "price": 3.0, "bashing": 1.1, "cutting": 1.1 },
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_melee.json]:
```suggestion
    "proportional": { "price": 6.0, "bashing": 1.2, "cutting": 1.2, "weight": .8 },
```

File: data/mods/Magiclysm/enchanted/enchanted_melee.json [hunk 2]
```diff
@@ -0,0 +1,758 @@
+[
+  {
+    "type": "GENERIC",
+    "id": "cudgel_plus_one",
+    "copy-from": "cudgel",
+    "looks_like": "cudgel",
+    "name": "cudgel +1",
+    "proportional": { "price": 3.0, "bashing": 1.1, "cutting": 1.1 },
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_melee.json]:
```suggestion
    "proportional": { "price": 3.0, "bashing": 1.1, "cutting": 1.1, "weight": 0.9 },
```

File: data/mods/Magiclysm/enchanted/enchanted_melee.json [hunk 3]
```diff
@@ -0,0 +1,758 @@
...
+    "relative": { "to_hit": 1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "cudgel_plus_two",
+    "copy-from": "cudgel",
+    "looks_like": "cudgel",
+    "name": "cudgel +2",
+    "proportional": { "price": 3.0, "bashing": 1.1, "cutting": 1.1 },
+    "relative": { "to_hit": 1 }
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_melee.json]:
```suggestion
    "relative": { "to_hit": 2 }
```

File: data/mods/Magiclysm/enchanted/enchanted_melee.json (hunk 1 above)
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_melee.json]: Damn, I should have caught that, it was the first weapon I did, but I only did the +1 version and went back and did the +2 at the end.

**Comment by Contemplator11** [comment]
> `SUMMARY: Mods "Basic magical weapons and enchanted item crafting framework for Magiclysm"`
> 
> Included is the most basic magical weapons, weapons +1 and +2.
> This covers almost all melee weapons, though I'm sure some were missed.
> The improvements from a standard weapon to a +1/+2 one are an equivilent increase in to_hit, a 10% increase in damage per +, and a 10% decrease in weight, thus improving attack speeds.
> 
> The framework for the rest of crafting adds an additional tab on the crafting screen at the end, called Enchanted.
> For the time being this is primarily a placeholder, but there will be recipes flowing into it as recipes are created to go with content items.
> Players will be able to craft their own +1/+2, and beyond, weapons and assorted magical gear.
> 
> Enchanting items will be very time consuming, as one would imagine.
> Be prepared to spend 24, 48, or even 72 hours and more of time working on items.

Weapons +1 and beyond are classical from DnD.
The system that explicitly allowed player characters craft magical items, DnD 3.5, required a pretty hefty amount of gold for each layer of enchantment. Do you plan to introduce some kind of magical components and magical tools of the trade?

**Comment by foulman** [comment]
> 
> 
> > `SUMMARY: Mods "Basic magical weapons and enchanted item crafting framework for Magiclysm"`
> > Included is the most basic magical weapons, weapons +1 and +2.
> > This covers almost all melee weapons, though I'm sure some were missed.
> > The improvements from a standard weapon to a +1/+2 one are an equivilent increase in to_hit, a 10% increase in damage per +, and a 10% decrease in weight, thus improving attack speeds.
> > The framework for the rest of crafting adds an additional tab on the crafting screen at the end, called Enchanted.
> > For the time being this is primarily a placeholder, but there will be recipes flowing into it as recipes are created to go with content items.
> > Players will be able to craft their own +1/+2, and beyond, weapons and assorted magical gear.
> > Enchanting items will be very time consuming, as one would imagine.
> > Be prepared to spend 24, 48, or even 72 hours and more of time working on items.
> 
> Weapons +1 and beyond are classical from DnD.
> The system that explicitly allowed player characters craft magical items, DnD 3.5, required a pretty hefty amount of gold for each layer of enchantment. Do you plan to introduce some kind of magical components and magical tools of the trade?

Yes, I have included new 3 tool properties, but how it all comes together is going to be something we work on. I don't plan on going nuts with random monster organ collection as far as making enchanted items goes, I think that's something @LaVeyanFiend is saving for his potions, I'm going to try to stick to rare elements, and introduce some materials of my own as things progress a little. Expect this to grow quite rapidly, and I'm happy to see people contribute items or ideas for them.

I will be scraping any and all RPG sources I can find for item ideas, flaming swords, Defenders, etc. It will be in there.

**Comment by Contemplator11** [comment]
> Great! This is just what I needed for the dungons I was working on -- more item drops!
> We can talk about exactly what the tool qualities are, but adding them in sooner rather than with the recipes will allow some more brainstorming.

I suggest that rare components for magic items are going to be precious stones (rubies, sapphires, diamonds, etc), since it's very common in DnD, and precious stones are already introduced into the game, and a hard to find.
I suggest that a basic tool necessary for ench
antment is a magic circle, which can be constructed as a furniture (* command). I suppose a basic magic circle would require spellcraft 3 (as a makeshift welder, if I remember correctly). As of materials the circle, I suppose it should be tied for a magic class:
- Magus/Animist: Salt;
- Kelvinist/Stormshaper - Wood Ash;
- Technomancer/Earthsaper: Permanent marker;
- Druid/Biomancer: Blood.

As for "item" tools, I suggest a "rune binding" quality, and an item for that would be a chisel for metal, knife for wooden items.
The idea is that once the runes are inscribed, the item must "saturate" with magical energy being placed inside the circle for days or even weeks. This process is already implemented for brewing wine and smoking meat, if I remember correctly.

Fire-based enchantments can require a knowledge of an Ild (fire) rune, frost-based - Ise (Ice) rune, lightning-based - Uvar (storm), and so on. I attach an image of runes from a recent DnD adventure.
https://i.pinimg.com/originals/fa/90/a8/fa90a809a21697b35aa38a361a58da02.jpg
Various runes can be added to existing spellbooks, or even scattered among holy books, for example, in Hávamál, since it's a scripture to Odin. :)

**Comment by foulman** [comment]
I already have 2 tools created, and a 3rd one planned, as well as use of Runes for various purposes, even hopefully as weapon mods. For the time being, I want to keep enchanting simple and stick to the precious metals rather than involve gemstones, I think there is a jewelry rework going on in another PR and I'd rather see how that pans out before I go too wild on the recipes I'd only have to adjust later.

I do find the idea of an enchanting circle/furniture interesting, and I'm going to put that on my list to evaluate after I get a few more things out of my backlog.

A lot is going to depend on player feedback and playtesting.

**Comment by foulman** [comment]
Right now I'm working on a variety of magical fist weapons, weapons for unarmed combat, which I promised several people on discord, I hope to get that up today.

**Comment by dkargin** [comment]
Maybe we should try to implement enchantment by fiddling with `item::damage_level`. Right now there are only 4 levels (4th level = fortified). Or add another field to item class, which works the similar way (participates in calculations for armor/damage). Anyway, it is much better than adding bunch of generic items for enchantment+1, and making another set of items with +2, and so on.

I mean I would rather think of a way to enchant existing game items. Like `keen skinned zombie corpse +2` or `vorpal spoon +1`

**Comment by KorGgenT** [comment]
I agree with wanting to have a more proper way of adding item attributes via enchantments, but this will be a stopgap in terms of having more content to be able to build on

**Comment by foulman** [comment]
I did originally want to do something like that, along the lines of making an item stronger with a welder/tailor's kit.
I'm going to try and do weapon mods to add/remove additional enchantments from weapons, VORPAL is just a flag after all, just like FLAMING. I'm just not sure what and how well "gun" mods will work in various situations, but that's a little ways off.

**Comment by Contemplator11** [comment]
Maybe guns should not be enchantable? This continues the "magic disrupts technology" theme, as in Dresden Files and Shadowrun. Besides, Magiclysm already has this, when installed bionics lower characters' mana maximum.

**Comment by foulman** [comment]
The only issue with that is that we are explicitly making technomancy from the start, there's a whole class for it. 
I do have a few already made, a fire bolt shooting "flamelock" rifle, the Thunderbuss that shoots lightning, etc.. powered with mana. 
As well as a technomagic converter that will attach to some handguns and make them fire bolts of mana rather than a bullet. It stops being a 'gun' at that point. At the moment there are no enchanted firearms in that sense, but I was looking at doing enchanted bullets/shotgun shells/arrows/bolts and the like.

**Pull request merged**